core/src/avm2/e4x.rs

```diff
@@ -475,36 +489,82 @@ impl<'gc> E4XNode<'gc> {
         let mut attribute_nodes = Vec::new();
 
         let attributes: Result<Vec<_>, _> = bs.attributes().collect();
-        for attribute in attributes? {
-            let key = AvmString::new_utf8_bytes(
-                activation.context.gc_context,
-                attribute.key.into_inner(),
-            );
-            let value_str = custom_unescape(&attribute.value, decoder)?;
+        for attribute in attributes.map_err(|_| malformed_element(activation))? {
+            let (ns, local_name) = parser.resolve_attribute(attribute.key);
+            let name =
+                AvmString::new_utf8_bytes(activation.context.gc_context, local_name.into_inner());
+            let namespace = match ns {
+                ResolveResult::Bound(ns) => Some(AvmString::new_utf8_bytes(
+                    activation.context.gc_context,
+                    ns.into_inner(),
+                )),
+                ResolveResult::Unknown(ns) if ns == b"xmlns" => continue,
+                // https://www.w3.org/TR/xml-names/#xmlReserved
+                // The prefix xml is by definition bound to the namespace name http://www.w3.org/XML/1998/namespace.
+                ResolveResult::Unknown(ns) if ns == b"xml" => {
+                    Some("http://www.w3.org/XML/1998/namespace".into())
+                }
+                ResolveResult::Unknown(ns) => {
+                    return Err(Error::AvmError(type_error(
+                        activation,
+                        &format!(
+                            "Error #1083: The prefix \"{}\" for element \"{}\" is not bound.",
+                            String::from_utf8_lossy(&ns),
+                            name
+                        ),
+                        1083,
+                    )?))
+                }
+                ResolveResult::Unbound => None,
+            };
+
+            let value_str = custom_unescape(&attribute.value, decoder)
+                .map_err(|_| malformed_element(activation))?;
             let value =
                 AvmString::new_utf8_bytes(activation.context.gc_context, value_str.as_bytes());
 
             let attribute_data = E4XNodeData {
                 parent: None,
-                local_name: Some(key),
+                namespace,
+                local_name: Some(name),
                 kind: E4XNodeKind::Attribute(value),
             };
             let attribute = E4XNode(GcCell::new(activation.context.gc_context, attribute_data));
             attribute_nodes.push(attribute);
         }
 
+        let (ns, local_name) = parser.resolve_element(bs.name());
+        let name =
+            AvmString::new_utf8_bytes(activation.context.gc_context, local_name.into_inner());
+        let namespace = match ns {
+            ResolveResult::Bound(ns) => Some(AvmString::new_utf8_bytes(
+                activation.context.gc_context,
+                ns.into_inner(),
+            )),
+            ResolveResult::Unknown(ns) => {
```

Comment:
Should this have the `xml` and `xmlns` cases as well?

Reply:
I am not sure if `<xmlns:dummy/>` is even valid XML, because `xmlns:` is used as an attribute usually, but when correctly binding the namespace like this it already works: `<x xmlns:xmlns="http://example.org"><xmlns:a></xmlns:a></x>`.

Interestingly Flash might have a bug related to `xml:`. `<xml:a />` works, but `<xml:a></xml:a>` throws
> The element type "a" must be terminated by the matching end-tag "&lt;/a>".

`<xml:a></a>` works! :exploding_head:

Reply:
So I added support for `xml`, but not `xmlns`, prefixes for elements.